lib/HLSL/DxilOutputColorBecomesConstant.cpp

```diff
@@ -150,18 +133,128 @@ bool DxilOutputColorBecomesConstant::runOnModule(Module &M)
 
   const hlsl::DxilSignature & OutputSignature = DM.GetOutputSignature();
 
+  Function * FloatOutputFunction = HlslOP->GetOpFunc(DXIL::OpCode::StoreOutput, Type::getFloatTy(Ctx));
+  Function * IntOutputFunction = HlslOP->GetOpFunc(DXIL::OpCode::StoreOutput, Type::getInt32Ty(Ctx));
+
+  bool hasFloatOutputs = false;
+  bool hasIntOutputs = false;
+
+  visitOutputInstructionCallers(FloatOutputFunction, OutputSignature, HlslOP, [&hasFloatOutputs](CallInst *) {
+    hasFloatOutputs = true;
+  });
+
+  visitOutputInstructionCallers(IntOutputFunction, OutputSignature, HlslOP, [&hasIntOutputs](CallInst *) {
+    hasIntOutputs = true;
+  });
+
+    if (!hasFloatOutputs && !hasIntOutputs)
+  {
+    return false;
+  }
+
+  // Otherwise, we assume the shader outputs only one or the other (because the 0th RTV can't have a mixed type)
+  assert(hasFloatOutputs || hasIntOutputs);
```

Comment:
Ha ha. Yes, you're right. I was thinking in human language: "or" usually means "either but not both" when we say it out loud :-)